app/src/main/java/org/thoughtcrime/securesms/database/Storage.kt

```diff
@@ -1371,14 +1371,17 @@ open class Storage(
         val threadDB = DatabaseComponent.get(context).threadDatabase()
         val groupDB = DatabaseComponent.get(context).groupDatabase()
         threadDB.deleteConversation(threadID)
-        val recipient = getRecipientForThread(threadID) ?: return
+
+        val recipient = getRecipientForThread(threadID)
+        if (recipient == null) {
+            Log.w(TAG, "Got null recipient when deleting conversation - aborting.");
+            return
+        }
+
         when {
-            recipient.isContactRecipient -> {
-                if (recipient.isLocalNumber) return
-                val contacts = configFactory.contacts ?: return
-                contacts.upsertContact(recipient.address.serialize()) { priority = PRIORITY_HIDDEN }
-                ConfigurationMessageUtilities.forceSyncConfigurationNowIfNeeded(context)
-            }
+            // Note: We don't do anything if the thread is a 1-on-1 and the recipient is a contact
+            // of ours (i.e., when recipient.isContactRecipient)
+
             recipient.isClosedGroupRecipient -> {
```

Comment:
maybe we can just early exit on this and save an indentation level. Then the comment applies more clearly.

```
// this is the comment
if (recipient.isClosedGroupRecipient) return
```

Reply:
Have cleaned up